provider/disk_utilities.py

```diff
@@ -0,0 +1,106 @@
+"""
+Disk utilities functions for managing disks and filesystems in guest environments.
+
+This module contains functions related to disk detection, filesystem creation,
+and command execution for various OS types.
+"""
+
+from virttest import utils_disk
+from virttest.utils_misc import get_linux_drive_path
+from virttest.utils_windows.drive import get_disk_props_by_serial_number
+
+
+def get_disk_by_id(os_type, disk_id, session):
```

Comment:
Hi @mcasquer 
I am a bit confused about the definition of the `disk` object by calling `get_disk_by_id`.
How do you define it? I think we are better to clarify what kind of information the user will get with the naming of the function. The concept `disk`  of `get_disk_by_id` compares with it returns is a bit of generalization 
Here, we can get the disk index inside Windows guest, and the disk path inside Linux guest.
So I think we could figure out and abstract out a name to represent such return value, for example: `get_disk_xxxx_by_id` to let the users know how and where they can call it instead of reading the code to get to know which information they will get.
Feel free to share your thoughts. Thank you!

Reply:
@YongxueHong what about `get_disk_reference_by_id()` ?